### PR TITLE
binutils-handle-windows-nul

### PR DESCRIPTION
**Pull request by himalr**
`S_ISREG` doesn't work correctly for Windows nul device.
No need to add `\\.\` prefix twice when accessing Windows nul device.

This fixes https://github.com/msys2/MINGW-packages/issues/14725

I'm not a C programmer and not that familiar with msys2 ecosystem. Please let me know if I missed something.

### CODE REVIEW

**Comment by lazka** [comment]
This needs to be reported upstream at least

**Comment by himalr** [comment]
> This needs to be reported upstream at least

You mean file a bug report at https://sourceware.org/bugzilla/ linking this pull request ? 

Or should I submit these patches to them ?

**Comment by lazka** [comment]
> You mean file a bug report at https://sourceware.org/bugzilla/ linking this pull request ?

Yes, ideally with a proper description of the problem, not just a link.

> Or should I submit these patches to them ?

That would be ideal, we don't want to maintain downstream patches if possible.

**Comment by himalr** [comment]
> That would be ideal, we don't want to maintain downstream patches if possible.

[Done](https://sourceware.org/bugzilla/show_bug.cgi?id=29947)

**Comment by lazka** [comment]
thanks!

**Comment by himalr** [comment]
The BFD patch has been applied to the [master branch](https://sourceware.org/git/gitweb.cgi?p=binutils-gdb.git;h=a8cf07d37b0063df888da82aed00babd1907ab7e). I'm gonna have to report the libiberty bug to GCC. Should I close this pull request along with the bug report ?

File: mingw-w64-binutils/PKGBUILD [hunk 1]
```diff
@@ -83,6 +87,13 @@ prepare() {
   # https://salsa.debian.org/mingw-w64-team/binutils-mingw-w64/-/tree/master/debian/patches
   patch -p2 -i "${srcdir}/reproducible-import-libraries.patch"
   patch -p2 -i "${srcdir}/specify-timestamp.patch"
+
+  # Handle Windows nul device
+  # https://github.com/msys2/MINGW-packages/issues/1840
+  # https://github.com/msys2/MINGW-packages/issues/10520
+  # https://github.com/msys2/MINGW-packages/issues/14725
+  patch -p1 -i "${srcdir}/libiberty-unlink-handle-windows-nul.patch"
```
Comment (lazka) [inline mingw-w64-binutils/PKGBUILD:95]:
```suggestion
  # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108276
  patch -p1 -i "${srcdir}/libiberty-unlink-handle-windows-nul.patch"
```

File: mingw-w64-binutils/PKGBUILD [hunk 2]
```diff
@@ -83,6 +87,13 @@ prepare() {
...
+  patch -p1 -i "${srcdir}/bfd-real-fopen-handle-windows-nul.patch"
```
Comment (lazka) [inline mingw-w64-binutils/PKGBUILD:96]:
```suggestion
  # https://sourceware.org/bugzilla/show_bug.cgi?id=29947
  patch -p1 -i "${srcdir}/bfd-real-fopen-handle-windows-nul.patch"
```

File: mingw-w64-binutils/PKGBUILD (hunk 1 above)
Comment (himalr) [reply mingw-w64-binutils/PKGBUILD:98]: I have applied those changes and made a new commit. I have also changed the hash values of the new patch files to lower-case and improved the libiberty patch. 

BTW, I mistakenly merged the upstream to this branch and had to rest all. I hope it won't be an issue.

**Comment by lazka** [comment]
Sorry for the delay

**Pull request merged**